astronomer/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -152,3 +158,12 @@ async def get_role_credentials(
                 )
                 return_response = response["Credentials"]
             return return_response
+
+    def _get_aioclient(self) -> AioBaseClient:
+        session = self.get_session(region_name=self.region_name, deferrable=True)
```

Comment:
Would this require bumping our minimum amazon provider version? I think the deferrable flag would have been added recently right?

Reply:
Not sure if it has all the needed pieces, I just extracted custom pieces and tailored in that PR to add support for refreshable credentials for only the `assume_role` method

Reply:
I think we should do a similar for other methods too which have been not covered